indexer.go

```diff
@@ -1,31 +1,77 @@
 package indexer
 
+import (
+	"github.com/pokt-foundation/pocket-go/provider"
+)
+
 // Provider interface of needed provider functions
 type Provider interface {
 	GetBlockHeight() (int, error)
-	// TODO: add missing functions
+	GetBlock(blockHeight int) (*provider.GetBlockOutput, error)
+	GetBlockTransactions(blockHeight int) (*provider.GetBlockTransactionsOutput, error)
 }
 
-// Writer interface of write function
-type Writer interface {
-	// TODO: add missing functions
+// Persistance layer interface (database, in-memory, etc.)
+type Persistance interface {
+	// Writes
+	WriteBlock(block *provider.GetBlockOutput) error
+	WriteTransactions(txs []*provider.Transaction) error
+	// Reads
+	ReadBlock(blockHeight int) (interface{}, error)
```

Comment:
I think we should enforce `map[string]interface{}` since all these will be key-value pairs